components/event-sources/reconciler/httpsource/reconcile_test.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2019 The Kyma Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package httpsource
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	k8stesting "k8s.io/client-go/testing"
+
+	"knative.dev/eventing/pkg/reconciler"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	rt "knative.dev/pkg/reconciler/testing"
+	"knative.dev/pkg/resolver"
+	fakeservingclient "knative.dev/serving/pkg/client/injection/client/fake"
+
+	fakesourcesclient "github.com/kyma-project/kyma/components/event-sources/client/generated/injection/client/fake"
+	. "github.com/kyma-project/kyma/components/event-sources/reconciler/testing"
+)
+
+const (
+	tNs      = "testns"
+	tName    = "test"
+	tImg     = "sources.kyma-project.io/http:latest"
+	tSinkURI = "http://" + tName + "-kn-channel." + tNs + ".svc.cluster.local"
+)
+
+func TestReconcile(t *testing.T) {
+	testCases := rt.TableTest{
+		/* Error handling */
+
+		{
+			Name:    "Source was deleted",
+			Key:     tNs + "/" + tName,
+			Objects: nil,
+			WantErr: false,
+		},
+		{
+			Name:    "Invalid object key",
+			Key:     tNs + "/" + tName + "/invalid",
+			WantErr: true,
```

Comment:
are we able to specify which errors to expect in such case?

Reply:
Unfortunately not, we would have to open a PR to add that feature to Knative's table tests: https://godoc.org/knative.dev/pkg/reconciler/testing#TableRow